recipes/catch2/3.x.x/conanfile.py

```diff
@@ -57,50 +61,68 @@ def _compilers_minimum_version(self):
+            if Version(self.settings.compiler.version) < minimum_version:
                 raise ConanInvalidConfiguration("{}/{}: Unsupported compiler: {}-{} "
                                                 "(https://github.com/p-ranav/structopt#compiler-compatibility)."
                                                 .format(self.name, self.version, self.settings.compiler, self.settings.compiler.version))
         else:
             self.output.warn("{}/{} requires C++14. Your compiler is unknown. Assuming it supports C++14.".format(self.name, self.version))
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["BUILD_TESTING"] = "OFF"
-        cmake.definitions["CATCH_INSTALL_DOCS"] = "OFF"
-        cmake.definitions["CATCH_INSTALL_HELPERS"] = "ON"
-        cmake.definitions["CATCH_CONFIG_PREFIX_ALL"] = self.options.with_prefix
-        if self.options.default_reporter:
-            cmake.definitions["CATCH_CONFIG_DEFAULT_REPORTER"] = self._default_reporter_str
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    # @functools.lru_cache(1)
+    # def _configure_cmake(self):
+    #     cmake = CMake(self)
+    #     cmake.definitions["BUILD_TESTING"] = "OFF"
+    #     cmake.definitions["CATCH_INSTALL_DOCS"] = "OFF"
+    #     cmake.definitions["CATCH_INSTALL_HELPERS"] = "ON"
+    #     cmake.definitions["CATCH_CONFIG_PREFIX_ALL"] = self.options.with_prefix
+    #     if self.options.default_reporter:
+    #         cmake.definitions["CATCH_CONFIG_DEFAULT_REPORTER"] = self._default_reporter_str
+    #     cmake.configure(build_folder=self._build_subfolder)
+    #     return cmake
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        for x in self.conan_data.get("patches", {}).get(self.version, []):
+            patch(**x)
+
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_TESTING"] = False
+        tc.variables["CATCH_INSTALL_DOCS"] = False
+        tc.variables["CATCH_INSTALL_HELPERS"] = True
+        tc.variables["CATCH_CONFIG_PREFIX_ALL"] = self.options.with_prefix
+        if self.options.default_reporter:
+            tc.variables["CATCH_CONFIG_DEFAULT_REPORTER"] = self._default_reporter_str
+        tc.generate()
 
     def build(self):
         self._patch_sources()
-        cmake = self._configure_cmake()
+        # cmake = self._configure_cmake()
+        # cmake.build()
+
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=os.path.join(self.source_folder, os.pardir))
```

Comment:
```suggestion
        cmake.configure()
```
with cmake_layout, it should work